src/reporter.ts

```diff
@@ -77,22 +81,36 @@ class RPReporter implements Reporter {
 
   onStdOut(chunk: string | Buffer, test?: TestCase, result?: TestResult): void {
     try {
-      const { type, data } = JSON.parse(String(chunk));
+      const { type, data, suite } = JSON.parse(String(chunk));
 
       switch (type) {
         case EVENTS.ADD_ATTRIBUTES:
-          this.addAttributes(data, test);
+          this.addAttributes(data, suite, test);
+          break;
+        case EVENTS.SET_DESCRIPTION:
+          this.setDescription(data, suite, test);
```

Comment:
Optional parameters are typically goes at the end of parameter list, so please update method signature to `this.setDescription(data, test, suite)`. The same for `this.addAttributes`.